src/doc/unstable-book/src/language-features/loop-break-value.md

```diff
@@ -4,7 +4,86 @@ The tracking issue for this feature is: [#37339]
 
 [#37339]: https://github.com/rust-lang/rust/issues/37339
 
+Documentation to be appended to section G of the book.
+
 ------------------------
 
+### Loops as expressions
+
+Like most things in Rust, loops are expressions; for example, the following is perfectly legal,
+if rather useless:
+
+```rust
+let result = for n in 1..4 {
+    println!("Hello, {}", n);
+};
+assert_eq!(result, ());
+```
+
+Until now, all the loops you have seen evaluate to either `()` or `!`, the latter being special
+syntax for "no value", meaning the loop never exits. A `loop` can instead evaluate to
+a useful value via *break with value*:
+
+```rust
+// Find the first square number over 1000:
+let mut n = 1;
+let square = loop {
+    if n * n > 1000 {
+        break n * n;
+    }
+    n += 1;
+};
+```
+
+The evaluation type may be specified externally:
+
+```rust
+// Declare that value returned is unsigned 64-bit:
+let n: u64 = loop {
+    break 1;
+};
+```
+
+It is an error if types do not agree, either between a "break" value and an external requirement,
+or between multiple "break" values:
+
+```rust
+loop {
+    if random_bool() {
+        break 1u32;
+    } else {
+        break 0u8;  // error: types do not agree
+    }
+};
+
+let n: i32 = loop {
+    break 0u32; // error: type does not agree with external requirement
+};
+```
+
+For now, breaking with a value is only possible with `loop`; the same functionality may
+some day be added to `for` and `while` (this would require some new syntax like
+`while f() { break 1; } default { break 0; }`).
```

Comment:
Your concerns are valid, but the why response is complex and depends on who you ask, imo it's too opinionated to get into that here. I definitely think your use of the loop term is correct, I just worry it may be confusing in this context.